src/analysis/rust_type.rs

```diff
@@ -191,11 +196,59 @@ fn rust_type_full(
             }
         }
         Custom(library::Custom { ref name, .. }) => Ok(name.clone()),
+        Function(ref f) => {
+            let full_name = type_id.full_name(&env.library);
+            if full_name == "Gio.AsyncReadyCallback" {
+                return Ok(format!("FnOnce(Result<(), Error>) + Send + 'static"));
+            }
+            if full_name == "GLib.DestroyNotify" {
+                return Ok(format!("Fn() + 'static"));
```

Comment:
Functions with callbacks with `scope="async"` and not GIO async functions are also `FnOnce`. `scope="call"` is `FnMut`, `scope="notified"` is `Fn`

Reply:
Ah this is for generating a Rust type for `GDestroyNotify`? That should never ever happen as such a thing wouldn't be safe

Reply:
Indeed, it's legacy code in the time when I allowed the destroy callback to be given by the users.